Spring-2019/linked_lists/SOLUTIONS_2019Spring.md

```diff
@@ -107,21 +88,41 @@ TODO :bug:
 
 TODO :bug:
 
-### 2. PROBLEM 2 TODO :bug:
-
-Source: TODO :bug:
+### 2. Delete Middle Node
 
-#### Naive/Simple Solution
-
-TODO :bug:
+Source: CTCI 6th edition 2.3
 
 #### Optimal Solution
 
```

Comment:
```suggestion
Because we don't have access to the head node of the list, we cannot cycle back to the node in question from the head of the list. This constrains our ability to safely delete the last node in the list, so checking that the node and the node's next item is not `null` is important. If these two checks pass, then we simply replace the node's value with it's successor's value and delete the successor node.
```